### PR TITLE
Piksi system daemon

### PR DESCRIPTION
**Pull request by gsmcmullin**
Add new `piksi_system_daemon`.  This new daemon process on Linux is responsible for implementing
system configuration settings for IPv4 Ethernet configuration and UART baudrates.

It also implements the new SBP shell command messages used for Piksi Multi firmware upgrade.  For firmware upgrade https://github.com/swift-nav/piksi_buildroot/pull/37 is also needed.

Do not merge until libsbp pointer has been updated:
https://github.com/swift-nav/libsbp/pull/412

### CODE REVIEW

File: package/piksi_system_daemon/piksi_system_daemon.mk [hunk 1]
```diff
@@ -0,0 +1,20 @@
+################################################################################
+#
+# sbp_settings_client
```
Comment (jacobmcnamee) [inline package/piksi_system_daemon/piksi_system_daemon.mk]: update name

File: package/piksi_system_daemon/src/main.c [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+u8 eth_ip_mode = IP_CFG_STATIC;
+static char eth_ip_addr[16] = "192.168.0.222";
+static char eth_netmask[16] = "255.255.255.0";
+static char eth_gateway[16] = "192.168.0.1";
+
+static void eth_update_config(void)
+{
+  system("ifdown eth0");
+
+  FILE *interfaces = fopen("/etc/network/interfaces", "w");
```
Comment (jacobmcnamee) [inline package/piksi_system_daemon/src/main.c]: Does appending to the file every time the setting is written work as expected?

File: package/piksi_system_daemon/src/main.c [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  return i > 1 ? i : 0;
+}
+
+int shell_command_output(zloop_t *loop, zmq_pollitem_t *item, void *arg)
+{
+  struct shell_cmd_ctx *ctx = arg;
+  msg_log_t *msg = alloca(256);
+  msg->level = 6;
+
+  if (raw_fgets(msg->text, 254, ctx->pipe) > 0) {
```
Comment (jacobmcnamee) [inline package/piksi_system_daemon/src/main.c]: `sizeof(msg->text)` ?

**Comment by jacobmcnamee** [comment]
Will this cause any problems for network boot in the dev config? Files are TFTP'd in another init script. Are the UARTs and network fully up and configured before the system daemon init script finishes?

File: package/piksi_system_daemon/src/main.c (hunk 3 above)
Comment (gsmcmullin) [inline package/piksi_system_daemon/src/main.c]: `msg->text` is a zero length array.  Can do `SBP_FRAMING_MAX_PAYLOAD_SIZE - sizeof(*msg)` if that's better.

File: package/piksi_system_daemon/src/main.c (hunk 2 above)
Comment (jacobmcnamee) [inline package/piksi_system_daemon/src/main.c]: Oh, yes it does. I guess as long as all the network interface settings (including possibly USB ethernet, for example) are contained in this daemon, we can just overwrite the whole file every time. Sounds fine.

**Comment by gsmcmullin** [comment]
TFTP boot config still works for me.  The network interface is configured by the kernel in this case, and the files are transferred before the daemon is started.  It is reconfigured by the daemon to the values in the settings, but only after the firmware and bitstream have been downloaded.

**Comment by jacobmcnamee** [comment]
Ok, sounds good. Is the default static IP also working?

**Comment by gsmcmullin** [comment]
Yes, the default configuration is static 192.168.0.222/24, and works.

**Comment by jacobmcnamee** [comment]
lgtm

**Pull request merged**